src/algebra/direct_sum/internal.lean

```diff
@@ -171,13 +172,28 @@ instance galgebra [add_monoid ι]
   (A : ι → submodule S R) [h : set_like.graded_monoid A] (s : S) :
     ↑(@direct_sum.galgebra.to_fun _ S (λ i, A i) _ _ _ _ _ _ _ s) = (algebra_map S R s : R) := rfl
 
+omit deq
 /-- A direct sum of powers of a submodule of an algebra has a multiplicative structure. -/
 instance nat_power_graded_monoid
   [comm_semiring S] [semiring R] [algebra S R] (p : submodule S R) :
   set_like.graded_monoid (λ i : ℕ, p ^ i) :=
 { one_mem := by { rw [←one_le, pow_zero], exact le_rfl },
   mul_mem := λ i j p q hp hq, by { rw pow_add, exact submodule.mul_mem_mul hp hq } }
 
+section homogeneous_element
```

Comment:
Why not move this down below `end submodule` so that you don't need the `_root_`.